misk-redis/src/main/kotlin/misk/redis/RealRedis.kt

```diff
@@ -302,6 +304,46 @@ class RealRedis(
     jedisPool.resource.use { jedis -> jedis.publish(channel, message) }
   }
 
+  override fun zadd(
+    key: String,
+    score: Double,
+    member: ByteString,
+    options: Set<String>): Long {
+    return jedisPool.resource.use { jedis ->
+      jedis.zadd(key.toByteArray(charset), score, member.toByteArray(), getZAddParams(options))}
+  }
+
+  override fun zadd(
+    key: String,
+    scoreMembers: Map<ByteString, Double>,
+    options: Set<String>): Long {
+    val params = getZAddParams(options)
+    val keyBytes = key.toByteArray(charset)
+    val scoreMembersBytes = scoreMembers.entries.associate { it.key.toByteArray() to it.value }
+
+    return jedisPool.resource.use { jedis -> jedis.zadd(keyBytes, scoreMembersBytes, params)}
+  }
+
+  override fun zscore(key: String, member: String): Double? {
+    return jedisPool.resource.use { it.zscore(key, member) }
+  }
+
+  private fun getZAddParams(options: Set<String>): ZAddParams {
+    val params = ZAddParams()
+
+    options.forEach {
+      when(it.uppercase()) {
+        ZAddOptions.XX.name -> params.xx()
+        ZAddOptions.NX.name -> params.nx()
+        ZAddOptions.LT.name -> params.lt()
+        ZAddOptions.GT.name -> params.gt()
+        ZAddOptions.CH.name -> params.ch()
+      }
+    }
```

Comment:
To throw an error if they provide an invalid param:
```suggestion
    val params = ZAddParams()

    options.forEach {
      when(ZAddOptions.valueOf(it.uppercase())) {
        ZAddOptions.XX -> params.xx()
        ZAddOptions.NX -> params.nx()
        ZAddOptions.LT -> params.lt()
        ZAddOptions.GT -> params.gt()
        ZAddOptions.CH -> params.ch()
      }
    }
```